frontend/awx/administration/instances/hooks/useInstanceToolbarActions.tsx

```diff
@@ -68,22 +49,58 @@ export function useInstanceToolbarActions(view: IAwxView<Instance>) {
               'You do not have permission to add instances. Please contact your organization administrator if there is an issue with your access.'
             ),
       },
-      {
+    ];
+    if (isK8s) {
+      const removeInstanceAction: IPageAction<Instance> = {
         type: PageActionType.Button,
-        isHidden: () => isK8s === false,
         selection: PageActionSelection.Multiple,
         icon: MinusCircleIcon,
         label: t('Remove instance'),
         onClick: (instance: Instance[]) => removeInstances(instance),
         isDisabled: (instances: Instance[]) => cannotRemoveInstances(instances, t),
         isDanger: true,
-      },
+      };
+      actions.push(removeInstanceAction);
+    }
+    actions = actions.concat(healthCheckAction);
+
+    return actions;
+  }, [t, canAddAndEditInstances, pageNavigate, isK8s, removeInstances, healthCheckAction]);
+}
+
+export function useRunHealthCheckToolbarAction(view: IAwxView<Instance>, isPinned?: boolean) {
+  const { t } = useTranslation();
+  const runHealthCheck = useRunHealthCheck(view.unselectItemsAndRefresh);
+
+  const cannotRunHealthCheckDueToPending = useCallback(
+    (instance: Instance) => {
+      if (instance.health_check_pending)
+        return t(
+          `Instance has pending health checks. Wait for those to complete before attempting another health check.`
```

Comment:
This never gets displayed, if the error message is returned, it will only fulfill the condition below and returns "Cannot run health checks on one or more of the selected instances".